### PR TITLE
sql,pgwire: fix timestamp and tz offset format

### PR DESCRIPTION
**Pull request by rafiss**
This change includes the following:
- Make data type size for time[stamp][tz] match Postgres in pg_catalog and over pgwire.
- Change parameter status sending so it occurs after sending CommandComplete.
- Normalize "UTC" and "GMT" to uppercase when storing timezone offset.

The new tests cover all of these.

Epic: None
Release note (bug fix): Adjusted the size reported for TIME, TIMETZ, TIMESTAMP, TIMESTAMPTZ, and DATE types in the pgwire protocol.

### CODE REVIEW

File: pkg/sql/pgwire/testdata/pgtest/timezone [hunk 1]
```diff
@@ -5,8 +5,8 @@ Query {"String": "SET TIME ZONE \"UTC\""}
 until
 ReadyForQuery
 ----
-{"Type":"ParameterStatus","Name":"TimeZone","Value":"UTC"}
 {"Type":"CommandComplete","CommandTag":"SET"}
+{"Type":"ParameterStatus","Name":"TimeZone","Value":"UTC"}
```
Comment (otan) [inline pkg/sql/pgwire/testdata/pgtest/timezone:9]: i have just discovered PG doesn't send this out if the value of the timezone didn't change, but that's not a now problem...

File: pkg/sql/pgwire/testdata/pgtest/timezone [hunk 2]
```diff
@@ -52,8 +52,8 @@ Query {"String": "SET TIME ZONE \"America/Chicago\""}
 until
 ReadyForQuery
 ----
-{"Type":"ParameterStatus","Name":"TimeZone","Value":"America/Chicago"}
 {"Type":"CommandComplete","CommandTag":"SET"}
+{"Type":"ParameterStatus","Name":"TimeZone","Value":"America/Chicago"}
```
Comment (otan) [inline pkg/sql/pgwire/testdata/pgtest/timezone:56]: i feel like some other test files need to change as a result of this change. how did they pass `pgwire` datadriven tests before o_o

**Review by otan — APPROVED** [review]
i think some other test files with ParamStatusUpdates will need to change but otherwise LGTM

File: pkg/sql/pgwire/testdata/pgtest/timezone (hunk 2 above)
Comment (rafiss) [reply pkg/sql/pgwire/testdata/pgtest/timezone:56]: this file wasn't passing against PG before 🤥 

let's see which others turn up

File: pkg/sql/exec_util.go [hunk 3]
```diff
@@ -2929,11 +2929,14 @@ func (m *sessionDataMutator) bufferParamStatusUpdate(param string, status string
...
 // SetApplicationName sets the application name.
 func (m *sessionDataMutator) SetApplicationName(appName string) {
+	oldName := m.data.ApplicationName
 	m.data.ApplicationName = appName
 	if m.onApplicationNameChange != nil {
 		m.onApplicationNameChange(appName)
 	}
-	m.bufferParamStatusUpdate("application_name", appName)
+	if oldName != appName {
+		m.bufferParamStatusUpdate("application_name", appName)
```
Comment (otan) [inline pkg/sql/exec_util.go:2938]: i wonder if we should make `bufferParamStatusUpdate` always take in the old value instead and do it all in one place.

Comment (otan) [reply pkg/sql/exec_util.go:2938]: hmm nvm, for `is_superuser` we always have to send that over.

**Review by otan — APPROVED** [review]

**Comment by rafiss** [comment]
tftr!

bors r=otan

**Pull request merged**